### ISSUE TITLE
dynamically plot_config

### ISSUE THREAD

**Issue by arris69** [freqtrade/freqtrade#5984]
## Describe your environment

  * latest git develop version
  
## 
hi,

for now i manipulate the plot_config dictionary from within my strategies __init__ function to get the resampled timeframes displayed.
is there a better way to do this for dynamically generated sample periods per design?

Examplecode:
`

    def get_ticker_short(self):
        print(f'Timeframe: {self.timeframe} {timeframe_to_minutes(self.timeframe)}')
        self._short = str(timeframe_to_minutes(self.timeframe) * 6)
        return (timeframe_to_minutes(self.timeframe) * 6)

    def get_ticker_long(self):
        self._long = str(self.get_ticker_short() * 7)
        return (self.get_ticker_short() * 7)

    def __init__(self, **kwargs):
        self.get_ticker_short()
        self.get_ticker_long()
        self.plot_config['subplots']['KDJ 1D'][f'resample_{self._short}_K'] = self.plot_config['subplots']['KDJ 1D'].pop('resample_short_K')
        self.plot_config['subplots']['KDJ 1D'][f'resample_{self._short}_D'] = self.plot_config['subplots']['KDJ 1D'].pop('resample_short_D')
        self.plot_config['subplots']['KDJ 1D'][f'resample_{self._short}_J'] = self.plot_config['subplots']['KDJ 1D'].pop('resample_short_J')
        self.plot_config['subplots']['KDJ 1D'][f'resample_{self._short}_rsi'] = self.plot_config['subplots']['KDJ 1D'].pop('resample_short_rsi')
        self.plot_config['subplots']['Forecast'][f'resample_{self._short}_macd'] = self.plot_config['subplots']['Forecast'].pop('resample_short_macd')
        self.plot_config['subplots']['Forecast'][f'resample_{self._short}_macdsignal'] = self.plot_config['subplots']['Forecast'].pop('resample_short_macdsignal')

        self.plot_config['subplots']['KDJ 1W'][f'resample_{self._long}_K'] = self.plot_config['subplots']['KDJ 1W'].pop('resample_long_K')
        self.plot_config['subplots']['KDJ 1W'][f'resample_{self._long}_D'] = self.plot_config['subplots']['KDJ 1W'].pop('resample_long_D')
        self.plot_config['subplots']['KDJ 1W'][f'resample_{self._long}_J'] = self.plot_config['subplots']['KDJ 1W'].pop('resample_long_J')
        self.plot_config['subplots']['KDJ 1W'][f'resample_{self._long}_rsi'] = self.plot_config['subplots']['KDJ 1W'].pop('resample_long_rsi')
        self.plot_config['subplots']['Forecast'][f'resample_{self._long}_macd'] = self.plot_config['subplots']['Forecast'].pop('resample_long_macd')
        self.plot_config['subplots']['Forecast'][f'resample_{self._long}_macdsignal'] = self.plot_config['subplots']['Forecast'].pop('resample_long_macdsignal')
        super().__init__(**kwargs)
`

thnx for suggestions/clarifying

**Comment by xmatthias** [comment]
one thing to note:
 you should always pass on both `*args` and `*kwargs`. We do not guarantee that instantiation will always happen with kwargs only - so a positional argument would be "lost" in your implementation.

I guess you can change plot_config to a property. It's not part of the documentation, and not really tested - but i don't see why it should not work.

```
@property
def plot_config(self):
    return {
        // ... whatever
    }
```


If this works for you, i'd encourage you to make a PR to update this in the documentation, as this seems like a pretty common question up late.

**Comment by arris69** [comment]
the @property solution is working fine.

thanx for the hint. but sorry, i hate to write documentations.

`
   
    @property
    def plot_config(self):
        """
            There are a lot of solutions how to build the dictonary, hardcoded (as one big dictonary or add the elements one by one)
            Example 1:
                plot_config = {'main_plot': {}, 'subplots': {}}

            Example 2:
                plot_config = {}
                plot_config['main_plot'] = {}
                plot_config['subplots'] = {}
                plot_config['main_plot']['sma'] = {}
                plot_config['main_plot']['ema3'] = {}
                plot_config['main_plot']['ema5'] = {'color': 'green'}
                plot_config['subplots']['Other'] = {'macd': {'color': 'red'}, 'macdsignal': {'color': 'blue'}}

            Example 3: (use functions. here we resample 4h candles to 1d and 1w).
                plot_config = {}
                plot_config['main_plot'] = {}
                plot_config['subplots'] = {}
                plot_config['main_plot']['ema50'] = {'color': 'green'}
                plot_config['main_plot']['ema200'] = {'color': 'red'}
                plot_config['subplots']['Forecast'] = {'macd': {'color': 'red'}, 'macdsignal': {'color': 'blue'}}
                # some kind of append to the section.
                plot_config['subplots']['Forecast'].update({f'resample_{timeframe_to_minutes(self.timeframe) * 6}_macd': {}, f'resample_{timeframe_to_minutes(self.timeframe) * 6}_macdsignal': {}})
                plot_config['subplots']['Forecast'].update({f'resample_{timeframe_to_minutes(self.timeframe) * 6 * 7}_macd': {}, f'resample_{timeframe_to_minutes(self.timeframe) * 6 * 7}_macdsignal': {}})
        """
        plot_config = {}
        plot_config['main_plot'] = {}
        plot_config['subplots'] = {}

        return plot_config

`

**Comment by xmatthias** [comment]
Closed in cf5ff9257d432d170af14c4881fc3d098caa817d.
